crates/oxc_linter/src/rules/unicorn/prefer_type_error.rs

```diff
@@ -468,5 +471,56 @@ fn test() {
         ",
     ];
 
-    Tester::new(PreferTypeError::NAME, pass, fail).test_and_snapshot();
+    let fix = vec![
+        (
+            r"if (!isFinite(foo)) { throw new Error(); }",
+            r"if (!isFinite(foo)) { throw new TypeError(); }",
+        ),
+        (
+            r"if (isNaN(foo) === false) { throw new Error(); }",
+            r"if (isNaN(foo) === false) { throw new TypeError(); }",
+        ),
+        (
+            r"if (Array.isArray(foo)) { throw new Error('foo is an Array'); }",
+            r"if (Array.isArray(foo)) { throw new TypeError('foo is an Array'); }",
+        ),
+        (
+            r"if (foo instanceof bar) { throw new Error(foobar); }",
+            r"if (foo instanceof bar) { throw new TypeError(foobar); }",
+        ),
+        (
+            r"if (_.isElement(foo)) { throw new Error(); }",
+            r"if (_.isElement(foo)) { throw new TypeError(); }",
+        ),
+        (
+            r"if (_.isElement(foo)) { throw new Error; }",
+            r"if (_.isElement(foo)) { throw new TypeError; }",
+        ),
+        (
+            r"if (wrapper._.isElement(foo)) { throw new Error; }",
+            r"if (wrapper._.isElement(foo)) { throw new TypeError; }",
+        ),
+        (
+            r"if (typeof foo == 'Foo' || 'Foo' === typeof foo) { throw new Error(); }",
+            r"if (typeof foo == 'Foo' || 'Foo' === typeof foo) { throw new TypeError(); }",
+        ),
+        (
+            r"if (Number.isFinite(foo) && Number.isSafeInteger(foo) && Number.isInteger(foo)) { throw new Error(); }",
+            r"if (Number.isFinite(foo) && Number.isSafeInteger(foo) && Number.isInteger(foo)) { throw new TypeError(); }",
+        ),
+        (
+            r"if (wrapper.n.isFinite(foo) && wrapper.n.isSafeInteger(foo) && wrapper.n.isInteger(foo)) { throw new Error(); }",
+            r"if (wrapper.n.isFinite(foo) && wrapper.n.isSafeInteger(foo) && wrapper.n.isInteger(foo)) { throw new TypeError(); }",
+        ),
+        (
+            r"if (wrapper.f.g.n.isFinite(foo) && wrapper.g.n.isSafeInteger(foo) && wrapper.n.isInteger(foo)) { throw new Error(); }",
+            r"if (wrapper.f.g.n.isFinite(foo) && wrapper.g.n.isSafeInteger(foo) && wrapper.n.isInteger(foo)) { throw new TypeError(); }",
+        ),
+        (
+            r"if (_.isElement(foo)) { throw (new Error()); }",
```

Comment:
Can you please add some test cases that use "strange" expressions for `call_expe.callee`, e.g. parenthesis, type casts, chain expressions, `globalThis.Error`, etc?

Reply:
i don't think we currently report violations for these cases. it might be worth doing in another pr as an enhancement?